mne_bids/tests/test_mne_bids.py

```diff
@@ -283,6 +297,15 @@ def test_set():
     cmd = ['bids-validator', '--bep006', output_path]
     run_subprocess(cmd, shell=shell)
 
+    # Also cover iEEG
+    # We use the same data and pretend that eeg channels are ecog
+    raw.set_channel_types({'EEG 000': 'ecog'})
```

Comment:
no it doesn't ... but I assume that EEG channels could theoretically even be recorded simultaneously with the same amplifier that is used for the ecog electrodes ... so an iEEG dataset with eeg channels is a bit like an EEG dataset with eog channels?

Not sure, but this seems benign to me :-)

Reply:
I'd rather not step into the territory of simultaneous recordings. It's quite complex. See here: https://github.com/bids-standard/bids-specification/issues/86 and my opinion is that it's not smoothed out yet. The only exception is MEG + EEG which is already part of the BIDS main spec.

Reply:
yeah, I am well aware of the simultaneous recording issue ... it's really tough to work something out. But before I continue to argue: what is your suggestion for this PR?

Reply:
basically don't allow any combination of simultaneous recordings other than EEG + MEG. And when testing you can just loop over all the channels and set them to iEEG instead of doing for just one.

Reply:
I'm fine supporting them if there are clear guidelines in the BIDS specification which BEP takes precedence when both are present. Otherwise, do we write them as `_eeg.<ext>` or `_ieeg.<ext>` ?

Reply:
My point is: iEEG + EEG = `_ieeg.<ext>` is something ad hoc for `mne-bids` or something BIDS endorses?

I don't know of any data that has ecog + meg, so you're good there I think :)

Reply:
it is completely ad hoc but in my opinion the only logical conclusion. A procedure where ecog electrodes are applied usually implies an open skull in some way or the other: Nobody would do that to obtain an EEG dataset.

However, it is quite easy to collect some EEG data, while performing an iEEG data collection.

Reply:
okay, moving forward: To do what you suggest, we would need to adjust `_handle_kind`. Currently it's something like `if 'ecog' in raw: kind='ieeg'`, but from the discussion I get that you actually want something like `if 'ecog' in raw: ... check that there are no eeg or meg channels ... if true: kind=ieeg ... if false: raise error("cannot unambiguously infer kind. please pass manually using parameter xyz")`

is that representing your standpoint?

Reply:
yes, essentially. But see if you can probably achieve it without a nested `if` clause :-)